src/core/tsi/ssl_transport_security.h

```diff
@@ -79,6 +82,7 @@ tsi_result tsi_create_ssl_client_handshaker_factory(
     const tsi_ssl_pem_key_cert_pair* pem_key_cert_pair,
     const char* pem_root_certs, const char* cipher_suites,
     const char** alpn_protocols, uint16_t num_alpn_protocols,
+    grpc_core::RefCountedPtr<grpc_core::SslSessionLRUCache> ssl_session_cache,
```

Comment:
Due to backward compatibility, let's not change this API. Keep this API as it its. But add a new API as follows. 

```
typedef struct {
   const tsi_ssl_pem_key_cert_pair* pem_key_cert_pair,
   const char* pem_root_certs,
   const char* cipher_suites,
   const char* pem_root_certs,
   const char* cipher_suites,
   const char** alpn_protocols,
   uint16_t num_alpn_protocols,
   tsi_ssl_session_lru_cache* cache,
} tsi_ssl_client_handshaker_options;

Create a new API called tsi_create_ssl_client_handshaker_factory_with_options(
  tsi_ssl_client_handshaker_options* options, 
  tsi_ssl_client_handshaker_factory** factory);
```